agent/agents/postgres/pgstatmonitor/pgstatmonitor.go

```diff
@@ -344,14 +353,32 @@ func (m *PGStatMonitorQAN) checkDefaultWaitTime(waitTime time.Duration) bool {
 type settings map[string]*pgStatMonitorSettingsTextValue
 
 func (m *PGStatMonitorQAN) getSettings() (settings, error) {
-	var settingsRows []reform.Struct
-
 	settingsValuesAreText, err := areSettingsTextValues(m.q)
 	if err != nil {
 		return nil, err
 	}
+
+	pgsmVersion, _, err := getPGMonitorVersion(m.q)
+	if err != nil {
+		return nil, err
+	}
+
+	var settingsRows []reform.Struct
 	if settingsValuesAreText {
-		settingsRows, err = m.q.SelectAllFrom(pgStatMonitorSettingsTextValueView, "")
+		if pgsmVersion >= pgStatMonitorVersion20PG12 {
+			// In case of PGSM 2.0 and above we need create view first.
+			_, errSettings := m.q.Exec(pgsm20SettingsViewQuery)
+			if errSettings != nil {
+				// If it is already exists we just igroning error, otherwise it return error.
+				if !strings.Contains(errSettings.Error(), "already exists") {
+					return nil, errSettings
+				}
+			}
+
+			settingsRows, err = m.q.SelectAllFrom(pgStatMonitorSettingsTextValue20View, "")
```

Comment:
maybe we can typecast to underlying pq error to get code? Or we could remove redundant if statement with
```
_, errSettings := m.q.Exec(pgsm20SettingsViewQuery)
if errSettings != nil && !strings.Contains(errSettings.Error(), "already exists") {
  // If it is already exists we just igroning error, otherwise it return error.
  return nil, errSettings
}
settingsRows, err = m.q.SelectAllFrom(pgStatMonitorSettingsTextValue20View, "")
```